Track Buddy/MotionManager.swift

```diff
@@ -21,41 +23,63 @@ class MotionManager: ObservableObject {
      */
     
     // Acceleration values
-    @Published var x: Double = 0.0
-    @Published var y: Double = 0.0
-    @Published var z: Double = 0.0
+    @Published private(set) var x: Double = 0.0
+    @Published private(set) var y: Double = 0.0
+    @Published private(set) var z: Double = 0.0
     
     // Max G force values
-    @Published var maxBraking: Double = 0.0
-    @Published var maxAcceleration: Double = 0.0
-    @Published var maxRight: Double = 0.0
-    @Published var maxLeft: Double = 0.0
+    @Published private(set) var maxBraking: Double = 0.0
+    @Published private(set) var maxAcceleration: Double = 0.0
+    @Published private(set) var maxRight: Double = 0.0
+    @Published private(set) var maxLeft: Double = 0.0
+    
+    // Parameters
+    private let deviceMotionUpdateInterval: TimeInterval = 1/100
+    private let pointStorageLimit = 300 // number of motion updates stored for tracer graph
+    
+    private(set) var recentPoints: Deque<CGPoint> = [] // TODO: think about thread safety?
+    
+    func pointPath(atScale factor: CGFloat) -> CGMutablePath {
+        var points: [CGPoint] = []
+        for point in recentPoints {
+            let x = point.x * factor
+            let y = point.y * factor
+            points.append(CGPoint(x: x, y: y))
```

Comment:
Can you `.map` with Deque? If so this could be simplified to `let points = recentPoints.map { CGPoint(x: $0.x, y: $0.y) }`

Reply:
Yeah map should work. I was trying to use map the first time around a while ago but was doing this weird thing where I was trying to mutate the original CGPoints, so that naturally didn’t work. Then just kinda forgot about it when I sat down to work on it today 😛